lib/plugins/index.js

```diff
@@ -6,36 +6,41 @@ const {validatePlugin, validateStep, loadPlugin, parseConfig} = require('./utils
 const pipeline = require('./pipeline');
 const normalize = require('./normalize');
 
-module.exports = (context, pluginsPath) => {
+module.exports = async (context, pluginsPath) => {
   let {options, logger} = context;
   const errors = [];
 
   const plugins = options.plugins
-    ? castArray(options.plugins).reduce((plugins, plugin) => {
-        if (validatePlugin(plugin)) {
-          const [name, config] = parseConfig(plugin);
-          plugin = isString(name) ? loadPlugin(context, name, pluginsPath) : name;
+    ? await castArray(options.plugins).reduce(async (plugins, plugin) => {
+      const plugins2 = await plugins;
```

Comment:
I wonder if there is a better name for `plugins2`? Maybe `normalizedPlugins` or `awaitedPlugins`?

Reply:
yep, this is definitely part of the clean up that is necessary once this is working :)